autotest.pm

```diff
@@ -346,7 +346,8 @@ sub runalltests {
 
     write_test_order();
 
-    for my $t (@testorder) {
+    for my $testindex (0 .. $#testorder) {
+        my $t        = $testorder[$testindex];
```

Comment:
Here's the change that broke run-time scheduling. If new items get added to `@testorder`, the original version will cycle through them while the index sequence will ignore them.